pkgs/tools/text/diffsitter/default.nix

```diff
@@ -0,0 +1,26 @@
+{ lib
+, fetchFromGitHub
+, rustPlatform
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "diffsitter";
+  version = "0.6.6";
+
+  src = fetchFromGitHub {
+    owner = "afnanenayet";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-I4DruNglY8iok+UN/nvlKoqQJgL+F9O+9S0+64lk7OE=";
+    fetchSubmodules = true;
```

Comment:
Please use the tree sitter from nixpkgs.

Reply:
I just asked upstream for help with that. Unless you have an idea. I'm guessing I would need to patch their `build.rs`.

Reply:
If that is the case I am fine merging with it vendored. However it would nice to leave a comment explaining that it fetches its own treesitter and doesn't support an out-of-tree version. Ideally including a link to an upstream issue.

Reply:
Can we symlink the sources from nixpkgs instead?

Reply:
symlinking just the source seems like a very minor benefit as people overriding would expect to override the source of just that package and probably wouldn't end up changing this package anyways. Plus any build changes wouldn't be preserved. I think if we can't link the built result into there there isn't much point.

Reply:
> symlinking just the source seems like a very minor benefit

No, it is not. We don't have the source twice in the cache and also only need to update the package once.

> people overriding would expect to override the source of just that package and probably wouldn't end up changing this package anyways.

You can't really override a package without looking at the package definition first. If someone is doing that than really anything can happen because not every package is the same and just overwriting src or patches is often enough but not always.

Reply:
After taking a closer look the project is vendoring forks of the projects. I don't trust upstream to keep them up to date and the removed rust files don't do any harm if they are left in the repo.

Reply:
So you are ok with leaving the vendor for now?

Reply:
No, I still don't really like it.

Reply:
What is the status of this thread? IIUC we are still waiting for removal of the vendored package is that right?

Reply:
for now maybe we can use an approach similar to #138151